### PR TITLE
Add Trino 416 release notes

### LINKED ISSUE #17135: Release notes for 415

### LINKED ISSUE #17290: Release notes for 416

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 416 release. Fixes #17290 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 28 Apr 2023

* #17262 ✅ rn ✅ docs
* #17092 ✅ rn ✅ docs
* #17241 ❌  rn ✅ docs
* #17266 ✅ rn ✅ docs
* #17269 ✅ rn ✅ docs
* #17270 ✅ rn ✅ docs
* #16802 ✅ rn ✅ docs
* #17277 ✅ rn ✅ docs
* #16424 ✅ rn ✅ docs
* #17250 ✅ rn ✅ docs
* #17271 ✅ rn ✅ docs

## 29 Apr 2023

* #16888 ✅ rn ✅ docs
* #17273 ✅ rn ✅ docs
* #17297 ✅ rn ✅ docs
* #17220 ✅ rn ✅ docs

## 30 Apr 2023

* #17298 ✅ rn ✅ docs
* #16167 ✅ rn ✅ docs
* #17306 ✅ rn ✅ docs

## 1 May 2023

* #16876 ✅ rn ✅ docs
* #17283 ✅ rn ✅ docs
* #16223 ✅ rn ✅ docs
* #17309 ✅ rn ✅ docs
* #17229 ✅ rn ✅ docs
* #17291 ✅ rn ✅ docs
* #17312 ✅ rn ✅ docs

## 2 May 2023

* #16869 ✅ rn ✅ docs
* #17292 ✅ rn ✅ docs
* #17305 ✅ rn ✅ docs
* #17320 ✅ rn ✅ docs
* #17315 ✅ rn ✅ docs
* #15872 ✅ rn ✅ docs
* #17324 ✅ rn ✅ docs
* #17321 ✅ rn ✅ docs
* #17304 ✅ rn ✅ docs

## 3 May 2023

* #17317 ✅ rn ✅ docs
* #17284 ✅ rn ✅ docs
* #17233 ✅ rn ✅ docs
* #17245 ✅ rn ✅ docs
* #17027 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-416.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Release 416 (3 May 2023)
+
+## General
+
+* Improve performance of partitioned `INSERT`, `CREATE TABLE AS .. SELECT`, and
+  `EXECUTE` statements when the source table statistics are missing or 
+  inaccurate. ({issue}`16802`)
+* Improve performance of `LIKE` expressions that contain `%`. ({issue}`16167`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-416.md:8]: should we say "use pattern matching with %" instead ?

**Review by mosabua — APPROVED** [review]
One minor question.. but essentially also good as it stands

File: docs/src/main/sphinx/release/release-416.md (hunk 1 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-416.md:8]: That sounds more convoluted? What's confusing about the current phrasing?

Alternatively: "Improve performance of LIKE patterns that contain %" ?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-416.md:8]: I like that last one best ..

**Review by mosabua — APPROVED** [review]

**Pull request merged**